lib/product.rb

```diff
@@ -0,0 +1,21 @@
+class Product
+  attr_reader :category, :name, :unit_price, :quantity
+  attr_accessor :hoard
+  
+  def initialize(category, name, unit_price, quantity)
+    @category = category
+    @name = name
+    @unit_price = unit_price
+    @quantity = quantity.to_i
+    @hoard = false
+  
+  end
+
+  def total_price
+    @unit_price * @quantity
+  end
+
```

Comment:
It looks like this is missing the `hoard` method that should flip the `@hoard` instance variable from `false` to `true` based off of interaction pattern